Feature QSPI: add disabled flag to format phases

0xc0170 opened the pull request:
To be able to skip any phase in the format, use bus width set to NONE. Previously this
was done via SIZE definitions, this however did not provide flexibility to skip instruction phase
or data (as all of them can be done via 0-4 lines).

This way any format phase can be skipped if bus width is set to NONE.

@SenRamakri Please review

0xc0170 commented:
/morph build

SenRamakri commented on hal/qspi_api.h:
```diff
@@ -49,7 +50,6 @@ typedef enum qspi_bus_width {
 /** Address size
  */
 typedef enum qspi_address_size {
-    QSPI_CFG_ADDR_SIZE_NONE,
```
Can I know why we are removing QSPI_CFG_ADDR_SIZE_NONE/QSPI_CFG_ALT_SIZE_NONE. Theoretically they can be none.

0xc0170 replied:
The description or commit should contain the reason:

> To be able to skip any phase in the format, use bus width set to NONE. Previously this
was done via SIZE definitions, this however did not provide flexibility to skip instruction phase
or data (as all of them can be done via 0-4 lines).

0xc0170 replied:
I am open to suggestions, I find current size none limiting, this is the property that each phase has. We can add boolean enabled for instance instead, or any other?

0xc0170 commented:
I rebased  this. and added one more commit with another suggestions - use disabled flag (enabled did not make sense as by default it should be enabled). Therefore if flag is set to true, the phase is skipped no matter what values the structure for that phase holds.

0xc0170 commented:
Rebased to resolve conflict, open for review

0xc0170 commented:
I cleaned this branch, added driver edits to conform to the hal definitions for disabled flag.

Tested locally to compile QSPI test example. Please review

0xc0170 commented:
/morph build

0xc0170 commented:
rebased to resolve a conflict with destination branch, restarting CI

/morph build

0xc0170 commented:
/morph test
